File: Fibonacci.py
def verifica_fibonacci(num):
    a, b = 0, 1
    while b < num:
        a, b = b, a + b
    if b == num:
        return True
    else:
        return False

num = int(input("Informe um número: "))
fibonacci = [0, 1]

while fibonacci[-1] < num:
    fibonacci.append(fibonacci[-1] + fibonacci[-2])

if num in fibonacci:
    print(f"O número {num} pertence à sequência de Fibonacci!")
else:
    print(f"O número {num} não pertence à sequência de Fibonacci!")

File: fibonacci.c
#include <stdio.h>

int main() {
    int num, ant1 = 0, ant2 = 1, fib = 0, encontrado = 0;

    printf("Informe um número: ");
    scanf("%d", &num);

    while (fib <= num) {
        if (fib == num) {
            encontrado = 1;
            break;
        }
        fib = ant1 + ant2;
        ant1 = ant2;
        ant2 = fib;
    }

    if (encontrado) {
        printf("%d pertence à sequência de Fibonacci.", num);
    } else {
        printf("%d não pertence à sequência de Fibonacci.", num);
    }

    return 0;
}
